reflex/model.py

```diff
@@ -40,9 +40,12 @@ def get_engine(url: Optional[str] = None):
         console.print(
             "[red]Database is not initialized, run [bold]reflex db init[/bold] first."
         )
+    echo_db_query = False
+    if conf.env == constants.Env.DEV and constants.SQLALCHEMY_ECHO:
```

Comment:
in my testing the check for dev mode is not really working, and the reason is because `reflex run --env prod` sets dev mode in the main process, but then it spins up a second process to start the prod backend, and that process reads the config from the file (and does not realize that it should be in prod mode).

i'm not going to block on this, because our configuration needs to be refactored anyway, to protect this scenario, but just calling it out here as it was noticed during testing.

Reply:
Filed #1373 to track this problem. No code change is needed here at this time.

Reply:
> in my testing the check for dev mode is not really working, and the reason is because `reflex run --env prod` sets prod mode in the main process, but then it spins up a second process to start the prod backend, and that process reads the config from the file (and does not realize that it should be in prod mode).
> 
> i'm not going to block on this, because our configuration needs to be refactored anyway, to protect this scenario, but just calling it out here as it was noticed during testing.

Thank You @masenf. I perfectly understand what you’ve highlighted. At the moment the configurations aren’t synchronised across all processes, reason for this issue.

Reply:
> Filed #1373 to track this problem. No code change is needed here at this time.

I know is more technical to fix, but I am willing to lay my hands on it. So, please kindly assign this issue to me.

Reply:
@SlamChillz I left a note over in the other issue describing the requirements. I think if you comment on that issue I'll be able to assign you, but you're not showing up in my list of choices currently.